### PR TITLE
clfft: Fix for aarch64

### PR DESCRIPTION
**Pull request by t-nojiri**
I made a fix to be able to install on aarch64.

### CODE REVIEW

**Review by adamjstewart — COMMENTED** [review]
It might be good to share this patch with the developers. Obviously it won't work in its current state, but they may be able to suggest a more robust patch. We might also want to do the same thing for the 32-bit build, even though 32-bit is rare these days.

File: var/spack/repos/builtin/packages/clfft/package.py [hunk 1]
```diff
@@ -20,6 +20,10 @@ class Clfft(CMakePackage):
     depends_on('opencl@1.2:')
     depends_on('boost@1.33.0:', when='+client')
 
+    # This patch is required until the following PRs are merged.
+    # https://github.com/clMathLibraries/clFFT/pull/229
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/clfft/package.py]: Should we just use this patch instead?

**Comment by t-nojiri** [comment]
I changed the patch provision conditions and added a comment because of the following PR.
https://github.com/clMathLibraries/clFFT/pull/229

File: var/spack/repos/builtin/packages/clfft/package.py (hunk 1 above)
Comment (t-nojiri) [reply var/spack/repos/builtin/packages/clfft/package.py]: Is that a patch I created?

Comment (adamjstewart) [reply var/spack/repos/builtin/packages/clfft/package.py]: What I'm saying is that the patch referenced in this PR is more robust than the simple patch you added. What we could do is use:
```python
patch('https://github.com/clMathLibraries/clFFT/commit/eea7dbc888367b8dbea602ba539eb1a9cbc118d9.patch',
      sha256='3148d5937077def301b30b913bc2437df869204fca1de4385ccd46e3b98b13aa', when='@2.12.2')
```

Comment (t-nojiri) [reply var/spack/repos/builtin/packages/clfft/package.py]: I understand.
I fix it to use the above patch.

**Review by adamjstewart — APPROVED** [review]

**Pull request merged**